packages-internal/test-utils/src/describeConformance.tsx

```diff
@@ -157,45 +114,40 @@ export function testComponentProp(
 ) {
   describe('prop: component', () => {
     it('can render another root component with the `component` prop', () => {
-      const { mount, testComponentPropWith: component = 'em' } = getOptions();
-      if (!mount) {
-        throwMissingPropError('mount');
+      const { render, testComponentPropWith: component = 'em' } = getOptions();
+      if (!render) {
+        throwMissingPropError('render');
       }
 
-      const wrapper = mount(React.cloneElement(element, { component }));
+      const { container } = render(React.cloneElement(element, { component }));
 
-      expect(findRootComponent(wrapper, component).exists()).to.equal(true);
+      expect(container.firstChild?.nodeType).to.eq(1);
+      expect(container.firstChild?.nodeName.toLowerCase()).to.eq(component);
     });
   });
 }
 
 /**
- * MUI components can spread additional props to a documented component.
+ * MUI components spread additional props to its root.
  */
 export function testPropsSpread(
   element: React.ReactElement<any>,
   getOptions: () => ConformanceOptions,
 ) {
   it(`spreads props to the root component`, () => {
     // type def in ConformanceOptions
-    const { inheritComponent, mount } = getOptions();
-    if (!mount) {
-      throwMissingPropError('mount');
-    }
+    const { render } = getOptions();
 
-    if (inheritComponent === undefined) {
-      throw new TypeError(
-        'Unable to test props spread without `inheritComponent`. Either skip the test or pass a React element type.',
-      );
+    if (!render) {
+      throwMissingPropError('render');
     }
 
     const testProp = 'data-test-props-spread';
     const value = randomStringValue();
 
-    const wrapper = mount(React.cloneElement(element, { [testProp]: value }));
-    const root = findRootComponent(wrapper, inheritComponent);
+    const { container } = render(React.cloneElement(element, { [testProp]: value }));
 
-    expect(root.props()).to.have.property(testProp, value);
+    expect(container.firstChild).to.have.attribute(testProp, value);
```

Comment:
I think that to make this test valid we need to use a similar approach to `testClassName` by passing a `testId` and then selecting that element to avoid problems where we need certain additional wrappers for the tests, like in this case: https://github.com/mui/mui-x/pull/12295/files#diff-9d9c553a774da8e7527c95c670405a6cfeba292cf0be9962179d58a21f24d426R39

Reply:
In general, for all the test, probably the safest approach would be to go with the same flow, instead of relying on `container.firstChild`. WDYT? 🤔

Reply:
> In general, for all the test, probably the safest approach would be to go with the same flow, instead of relying on container.firstChild. WDYT?

Agree 100%, I'll refactor it

Reply:
Done ✅